test/integration/custom_cluster_integration_test.cc

```diff
@@ -0,0 +1,111 @@
+#include "envoy/api/v2/eds.pb.h"
+
+#include "common/network/address_impl.h"
+#include "common/upstream/load_balancer_impl.h"
+
+#include "test/config/utility.h"
+#include "test/integration/clusters/cluster_factory_config.pb.h"
+#include "test/integration/http_integration.h"
+
+namespace Envoy {
+namespace {
+
+const int UpstreamIndex = 0;
+const std::string CUSTOM_CLUSTER_CONFIG = R"EOF(
```

Comment:
Do you need a full config here? Can you just use the default config and a config modifier and modify the type of the initial cluster?

Reply:
I'm putting the full config here to make sure the deserialization work end to end.

Reply:
Sorry what do you mean?

Reply:
Ping on this. I'm still not sure why we need the whole config here.

/wait

Reply:
My understanding is that if I do the full config, it would test the end to end deserialization of the entire config.  If I use a config modifier, it's going to create the object in memory, then serialize and deserialize again.  There can be subtle differences, for example if someone introduce a breaking change an optional int field to a required field, then the current approach will break as expect, but scenario 2 might not.

Reply:
Sorry I still don't follow. I think in this case using a config modifier on top of the stock config should be fine and what we do in other similar tests? If I am missing something please explain further. :)